### PR TITLE
Added json filter #458

### PR DESCRIPTION
**Pull request by snoack**
See #458

### CODE REVIEW

**Comment by jeffwidman** [comment]
LGTM. While I'm not sure this is useful for Flask (I'd suggest `marshmallow` instead), I do think it makes sense for the Ansible project and probably a few others. 

Thanks for adding tests. Be good to add a note to the docs (including an example of how to pretty print the json) as well as the changelog. However, might wait for another maintainer or two to chime in whether they're on board with adding this filter. They may have objections that I haven't thought about.

**Comment by sciyoshi** [comment]
:+1: for the feature. One note is that any built-in JSON filter should properly handle escaping characters like `<`, `>`, and `&` so that it can safely be used inside `<script>` tags. (See Flask's code here: https://github.com/pallets/flask/blob/master/flask/json.py#L161)

File: jinja2/filters.py [hunk 1]
```diff
@@ -914,6 +914,14 @@ def do_rejectattr(*args, **kwargs):
     """
     return _select_or_reject(args, kwargs, lambda x: not x, True)
 
+def do_json(value, **kwargs):
+    """Serializes an object as JSON. Optionally given keyword arguments
+    are passed to json.dumps(), ensure_ascii however defaults to False.
+    """
+    import json
```
Comment (ThiefMaster) [inline jinja2/filters.py]: Not a big fan of this. Now you have a `json` filter that always uses the stdlib `json` module with the default settings and absolutely no way (except defining your own filter) to override this.

**Comment by ThiefMaster** [comment]
Besides `json` vs `tojson` (the latter is already used e.g. in Flask) I think it would also make this an `environmentfilter` and allow the environment to specify another json encoder.

File: jinja2/filters.py [hunk 2]
```diff
@@ -914,6 +914,14 @@ def do_rejectattr(*args, **kwargs):
...
+    kwargs.setdefault('ensure_ascii', False)
```
Comment (ThiefMaster) [inline jinja2/filters.py]: What's the advantage of disabling `ensure_ascii`? In Flask it defaults to `True`...

**Pull request closed without merging**